sky/packages/sky/lib/src/widgets/hero_controller.dart

```diff
@@ -97,7 +98,7 @@ class HeroController {
     Map<Object, HeroHandle> heroesTo = Hero.of(context, mostValuableKeys);
     _to.offstage = false;
 
-    PerformanceView performance = _to.performance;
+    PerformanceView performance = _to.performance.isAnimating ? _to.performance : _from.performance;
```

Comment:
This isn't right. They could both be animating when you start the transition. e.g. suppose the stocks app initial load transition took longer than the single stock view transition, and that the stock view was not entirely opaque (e.g. iOS-style translucency). open app, hit row, wait for that to finish, hit back. Now _to is animating, but you still want _from here.